locals.tf

```diff
@@ -9,37 +9,39 @@ locals {
 }
 
 locals {
-  locations_cached_or_live        = data.local_file.locations.content
-  regions_by_display_name         = { for v in local.regions_recommended_or_not : v.display_name => v }
-  regions_by_name                 = { for v in local.regions_recommended_or_not : v.name => v }
-  regions_by_name_or_display_name = merge(local.regions_by_display_name, local.regions_by_name)
-  regions_data_merged = [
-    for v in jsondecode(local.locations_cached_or_live).value :
-    merge(
-      {
-        name               = v.name
-        display_name       = v.displayName
-        paired_region_name = try(one(v.metadata.pairedRegion).name, null)
-        geography          = v.metadata.geography
-        geography_group    = v.metadata.geographyGroup
-        recommended        = v.metadata.regionCategory == "Recommended"
-      },
-      {
-        zones = sort(lookup(local.regions_to_zones_map, v.displayName, []))
-      }
-    ) if v.metadata.regionType == "Physical"
+  default_node_pool_zones = setsubtract(local.zones, local.restrictions)
+  filtered_vms = [
+    for item in data.azapi_resource_list.example.output.value :
+    item if(item.resourceType == "virtualMachines" && item.name == "Standard_D4d_v5")
   ]
-  # Filter out regions that are not recommended
-  regions_recommended_or_not          = [for v in local.regions_data_merged : v if v.recommended]
-  regions_to_zones_map                = { for v in local.regions_zonemappings : v.location => v.zones }
-  regions_zonemappings                = flatten([for v in jsondecode(local.regions_zonemappings_cached_or_live).resourceTypes : v.zoneMappings if v.resourceType == "virtualMachines"])
-  regions_zonemappings_cached_or_live = data.local_file.compute_provider.content
+  restrictions = try(local.filtered_vms[0].restrictions[0].restrictionInfo.zones, [])
```

Comment:
```suggestion
  restricted_zones = try(local.filtered_vms[0].restrictions[0].restrictionInfo.zones, [])
```

Reply:
i was just looking at this api https://learn.microsoft.com/en-us/rest/api/compute/resource-skus/list?view=rest-compute-2024-07-01&tabs=HTTP#resourceskurestrictioninfo and it seems to me this provides the list of restricted zones. We shouldn't merge it with the zones provided because as per the definition `List of availability zones where the SKU is restricted.`  this would mean we are merging restricted zones with zones that work.

Please correct me if i'm wrong about my understanding of this value, this is how we use it 

`default_node_pool_available_zones = setsubtract(local.zones, local.restrictions)`